modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/endpoint/IngestRestService.java

```diff
@@ -1036,7 +1036,10 @@ private Response ingestZippedMediaPackage(HttpServletRequest request, String wdI
   @POST
   @Produces(MediaType.TEXT_HTML)
   @Path("ingest/{wdID}")
-  @RestQuery(name = "ingest", description = "Ingest the completed media package into the system, retrieving all URL-referenced files, and starting a specified workflow",
+  @RestQuery(name = "ingest",
+             description = "<p>Ingest the completed media package into the system, retrieving all URL-referenced files, and "
```

Comment:
```
…retrieving all URL-referenced files…
```

I don't think that's true. Is it? URLs should just be referenced in the media package. They may be retrieved in some cases if a snapshot of the media package is created or if the assets are being used for processing. I would just leave that out.

Reply:
You are essentially saying if the workflow referenced by `wdID` is not touching the elements, they won't be downloaded. Tested that and seems to be true. The IngestDownload WOH does that after the WF has been instantiated in the test output log below. Will correct that.

```
(IngestServiceImpl:724) - Start adding track 326598d4-8f8f-4804-86d6-dc29c6db4ed0 from URL https://upload.wikimedia.org/wikipedia/commons/c/c0/Big_Buck_Bunny_4K.webm on mediapackage afb69cad-f363-4a2f-980b-82a5e517b25c
(IngestServiceImpl:1668) - Adding element of type Track to mediapackage afb69cad-f363-4a2f-980b-82a5e517b25c
(IngestServiceImpl:731) - Adding Tag:  to Element: 326598d4-8f8f-4804-86d6-dc29c6db4ed0
(IngestServiceImpl:737) - Successful added track 326598d4-8f8f-4804-86d6-dc29c6db4ed0 on mediapackage afb69cad-f363-4a2f-980b-82a5e517b25c at URL https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/326598d4-8f8f-4804-86d6-dc29c6db4ed0/Big_Buck_Bunny_4K.webm
(IngestServiceImpl:1170) - Starting a new workflow with ingested mediapackage afb69cad-f363-4a2f-980b-82a5e517b25c based on workflow definition '0-ng-schedule-and-upload-llz'
(IngestServiceImpl:1547) - Ingested mediapackage afb69cad-f363-4a2f-980b-82a5e517b25c is processed using workflow template '0-ng-schedule-and-upload-llz', specified during ingest
(IngestServiceImpl:1339) - No capture event found for id afb69cad-f363-4a2f-980b-82a5e517b25c
(IngestServiceImpl:1189) - Starting new workflow with ingested mediapackage 'afb69cad-f363-4a2f-980b-82a5e517b25c' using the specified template '0-ng-schedule-and-upload-llz'
(AbstractAssetManager:242) - Creating new version 0 of media package afb69cad-f363-4a2f-980b-82a5e517b25c
(AssetManagerWithMessaging:211) - Send update message for snapshot afb69cad-f363-4a2f-980b-82a5e517b25c, 0 to ActiveMQ
(WorkflowServiceImpl:807) - [>c67d5a04] Scheduling workflow 107380 for execution
(ExportWorkflowPropertiesWOH:87) - Start exporting workflow properties for workflow Workflow {107380}
(ImportWorkflowPropertiesWOH:97) - Cannot import workflow properties - no element with flavor 'processing/defaults' found in media package 'afb69cad-f363-4a2f-980b-82a5e517b25c'
(ExportWorkflowPropertiesWOH:138) - Added properties from Workflow {107380} as Attachment with flavor processing/defaults
(IncludeWorkflowOperationHandler:105) - Insert workflow partial-ingest into the current workflow instance
(IngestDownloadWorkflowOperationHandler:154) - Skipping downloading already existing element https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/326598d4-8f8f-4804-86d6-dc29c6db4ed0/Big_Buck_Bunny_4K.webm
(IngestDownloadWorkflowOperationHandler:186) - Downloaded the external element https://upload.wikimedia.org/wikipedia/commons/8/88/Big_Buck_Bunny_alt.webm
(IngestDownloadWorkflowOperationHandler:154) - Skipping downloading already existing element https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/c00adf43-842a-4e5b-b6b4-687f5d39381a/dublincore.xml
(IngestDownloadWorkflowOperationHandler:154) - Skipping downloading already existing element https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/f8dd407e-59e3-4e16-99c5-debe4ff3f679/processing_properties.xml
(IngestDownloadWorkflowOperationHandler:226) - Unable to delete external URI https://upload.wikimedia.org/wikipedia/commons/8/88/Big_Buck_Bunny_alt.webm, no working file repository found
(InspectWorkflowOperationHandler:152) - Inspecting track '326598d4-8f8f-4804-86d6-dc29c6db4ed0' of afb69cad-f363-4a2f-980b-82a5e517b25c
(MediaInspectionServiceRemoteImpl:126) - Enriching https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/326598d4-8f8f-4804-86d6-dc29c6db4ed0/big_buck_bunny_4k.webm using a remote media inspection service
(MediaInspectionServiceRemoteImpl:134) - Completing inspection of media file at https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/326598d4-8f8f-4804-86d6-dc29c6db4ed0/Big_Buck_Bunny_4K.webm using a remote media inspection service
(InspectWorkflowOperationHandler:152) - Inspecting track '326578d4-8f8f-4804-86d6-dc2916db4ed0' of afb69cad-f363-4a2f-980b-82a5e517b25c
(MediaInspectionServiceRemoteImpl:126) - Enriching https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/326578d4-8f8f-4804-86d6-dc2916db4ed0/big_buck_bunny_alt.webm using a remote media inspection service
(MediaInspectionServiceRemoteImpl:134) - Completing inspection of media file at https://opencast.example.com:8443/files/mediapackage/afb69cad-f363-4a2f-980b-82a5e517b25c/326578d4-8f8f-4804-86d6-dc2916db4ed0/Big_Buck_Bunny_alt.webm using a remote media inspection service
(SeriesWorkflowOperationHandler:196) - No series set, skip operation
(TagWorkflowOperationHandler:92) - Retagging mediapackage elements
(TagWorkflowOperationHandler:92) - Retagging mediapackage elements
```